scania-digital-design-system/tegel
# feat(tegel-styles): migration in monorepo, connection to foundations

## **Describe pull-request**  
Migrating tegel-styles repo as part of tegel monorepo.

What was done:
 - all foundation files are (colors, typography, spacing, grid, logotype) now on top level of monorepo
 - tegel-styles are renamed to styles and moved under packages
 - styles output different CSS files depending on user needs. User can choose to import all foundation file in one file or add the one that they want to use.
 - ag-grid overwrites is also a separate file so user can import only that
 - There is a node script that converts all scss to CSS.
 - Some files are libraries are removed as they are not used or needed anymore
 
My thinking was to enable modularity and flexibility as some users will consume @scania/tegel that has all styles so they just need ag-grid. Others might use just ag-grid and foundation files in case they do not use our @scania/tegel package. Also, users that would like to use just foundations, now they can do it.


## **Issue Linking:**  
- **Jira:** Add ticket number after `CDEP-`: [CDEP-3427](https://tegel.atlassian.net/browse/CDEP-3427)    

## **How to test**  
1. Run branch on your local
2. Check if build command on packages/styles works for you
3. Check if main repo build as usual



[CDEP-3427]: https://tegel.atlassian.net/browse/CDEP-3427?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
